### PR TITLE
Controller spec

### PR DESCRIPTION
**Pull request by Dadadon**
**
> Added RSpec as a gem to the project.
**
Created a Request spec file for controllers.
**
> For each action I checked:
**

- If the response status was correct.
- If a correct template was rendered.
- If the response body includes the correct placeholder text.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -129,6 +129,12 @@ To run tests, run the following command:
   bin/rails test test/models/article_test.rb
 ```
 
+To run tests on the views, run the following command:
+
+```sh
+  bin/rails bundle exec rspec spec/views
+```
+
```
Comment (GzimAsani) [inline README.md:137]: - [ ] I can see that you added all the necessary tests and they should pass, but unfortunately you have some problems with gemfile and gems therefore I am unable to run rspec, kindly fix it:
image
image

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### Hello, @Dadadon                    :wave:   
 Good job so far,  done There are some issues that you still need to work on to go to the next project but you are almost there!
 
 ## Highlights
 * Good job adding all the necessary tests +1: 

 ## Required Changes 

- [ ] Check for the comments below :arrow_double_down: 

 ## Optional suggestions
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- N/A



Cheers and Happy coding! clapclapclap
 
 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.
 
 **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/post_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+  it 'I can see a section for pagination if there are more posts than fit on the view.' do
+    expect(page).to have_content('Pagination')
+  end
+
+  it 'When I click on a post, it redirects me to that posts show page.' do
+    click_link "Post ##{@post1.id}"
+    expect(page).to have_current_path(user_post_path(@user1, @post1))
+  end
+end
```
Comment (Qoosim) [inline spec/views/post_index_spec.rb:69]: - [ ] Great work so far. However, there're few things to do before your project could be approved. According to the project requirements, you're expect to write test that ensures correct view template is rendered for both the index and show for the post controller.


```
 expect(response).to render_template(:index)

```

- [ ] Your test cases should include a test  that ensures the response body for the correct placeholder text. 

```
   Example :
      expect(response.body).to include("Welcome to post index page")

```

### Project Requirements

test

File: spec/views/post_show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  end
+
+  it 'I can see the name of each commentor.' do
+    expect(page).to have_content(@comment1.author.name)
+  end
+
+  it 'I can see each comment that was left.' do
+    expect(page).to have_content("#{@comment1.author.name}: #{@comment1.text}")
+  end
+end
```
Comment (Qoosim) [inline spec/views/post_show_spec.rb:43]: - [ ] According to the project requirements, you're expect to write test that ensures correct view template is rendered for both the index and show for the post controller.


```
 expect(response).to render_template(:show)

```

- [ ] Your test cases should include a test  that ensures the response body for the correct placeholder text. 

```
Example :
expect(response.body).to include("Welcome to post show  page")

```

### Project Requirements

test

File: spec/views/user_index_spec.rb [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    expect(page).to have_content("Number of posts: #{@user1.posts_counter}")
+    expect(page).to have_content("Number of posts: #{@user2.posts_counter}")
+    expect(page).to have_content("Number of posts: #{@user3.posts_counter}")
+  end
+
+  it 'When I click on a user, I am redirected to that user\'s show page' do
+    click_link @user1.name
+    expect(page).to have_current_path(user_path(@user1))
+  end
+end
```
Comment (Qoosim) [inline spec/views/user_index_spec.rb:46]: - [ ] According to the project requirements, you're expect to write test that ensures correct view template is rendered for both the index and show for the post controller.


```
expect(response).to render_template(:index)

```

- [ ] Your test cases should include a test  that ensures the response body for the correct placeholder text. 

```
Example :
   expect(response.body).to include("Welcome to user index  page")

```

### Project Requirements

test

File: spec/views/user_show_spec.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+  it 'I can see a button that lets me view all of a user\'s posts.' do
+    expect(page).to have_content('Show All Posts')
+  end
+
+  it "When I click to see all posts, it redirects me to the user's post's index page." do
+    click_link 'Show All Posts'
+    expect(page).to have_current_path(user_posts_path(@user1))
+  end
+end
```
Comment (Qoosim) [inline spec/views/user_show_spec.rb:51]: - [ ] According to the project requirements, you're expect to write test that ensures correct view template is rendered for both the index and show for the post controller.


```
expect(response).to render_template(:show)

```

- [ ] Your test cases should include a test  that ensures the response body for the correct placeholder text. 

```
Example :
expect(response.body).to include("Welcome to user show  page")

```

### Project Requirements

test

**Review by Qoosim — CHANGES_REQUESTED** [review]
#  *Status: Required Changes* :recycle: 

Hi @Dadadon, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


## Highlights
-  No linter errors. :heavy_check_mark: 
-  Good README file. :heavy_check_mark: 
-  Correct GitHub. :heavy_check_mark: 



 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by MahmoudBakr23 — APPROVED** [review]
Hi @Dadadon 😃, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations! 🎉

**Highlights**
- Well done on adding the necessary integration tests for your controllers

Cheers, and Happy coding!👏👏👏

Feel free to mention @MahmoudBakr23 to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by DammyShittu — COMMENTED** [review]
Hi @Dadadon,

## APPROVED 🟢

## Highlights
✔️ Passing tests

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joel-kalema — COMMENTED** [review]
Hi @Dadadon 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights
Linters are passing 👍 
The project is running successfully 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**